base/path.jl

```diff
@@ -613,3 +613,30 @@ relpath(path::AbstractString, startpath::AbstractString) =
 for f in (:isdirpath, :splitdir, :splitdrive, :splitext, :normpath, :abspath)
     @eval $f(path::AbstractString) = $f(String(path))
 end
+
+"""
+    uripath(path::AbstractString)
+
+Encode `path` as a URI as per [RFC1738](https://www.rfc-editor.org/rfc/rfc1738),
+[RFC3986](https://www.rfc-editor.org/rfc/rfc3986), and the [Freedesktop File URI
+spec](https://www.freedesktop.org/wiki/Specifications/file-uri-spec/).
+
+## Examples
+
+```julia-repl
+julia> uripath("/home/user/example file.jl")
+"file://<hostname>/home/user/example%20file.jl"
+```
+"""
+function uripath(path::String)
+    percent_escape(s) =
+        '%' * join(map(b -> string(b, base=16), codeunits(s)), '%')
+    encode_uri_component(s) =
+        replace(s, r"[^A-Za-z0-9\-_.~]+" => percent_escape)
+    string("file://", gethostname(), '/',
+           join(map(encode_uri_component,
+                    split(abspath(path), path_separator_re, keepempty=false)),
```

Comment:
Maybe this can be unnested a bit. Variables are free :P.

Reply:
Well... I suppose I could splurge and add one :stuck_out_tongue: